electron/execution.js

```diff
@@ -126,52 +126,70 @@ async function recordJourneys(data, browserWindow) {
   }
 }
 
-async function onTest(data) {
-  const result = {
-    succeeded: 0,
-    failed: 0,
-    skipped: 0,
-    journey: {},
-  };
-
+async function onTest(data, browserWindow) {
   const parseOrSkip = chunk => {
-    try {
-      return JSON.parse(chunk);
-    } catch (_) {
-      return {};
-    }
+    // at times stdout ships multiple steps in one chunk, broken by newline,
+    // so here we split on the newline
+    return chunk.split("\n").map(subChunk => {
+      try {
+        return JSON.parse(subChunk);
```

Comment:
I assume we'll only fall into `catch` because there will be a trailing newline and that won't be parsed, is that correct?

If so, what do you think of trying to `trim` or `filter` that and then having a `logger.error` call on the catch rather than defaulting to an empty `catch`?

Not a blocking comment btw, just an idea.

Reply:
Well, we're also interfacing with `stdout` which could potentially publish something un-parseable, which I think was the original motivation for including the `try...catch` here. It's not introduced in this PR AFAIK, just moved to a new spot. cc @kyungeunni

Reply:
I second what @justinkambic mentioned, chunk could contain un-parseable data. I personally would love to update this part of code a bit cleaner and safer (e.g. what if the object was too big so it is delivered via two separated chunks?) but I don't think we can get rid of try-catch in any case due to the reason mentioned above.

Reply:
Good points, thank you! 🙌